### PR TITLE
Updated README with Swift Package support instructions.

### PR DESCRIPTION
**Pull request by rr-james-hickman**
Added instructions per request in #73

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,11 +68,61 @@ When using **Github deploy keys**, GitHub servers will accept the _first_ known
 
 To support picking the right key in this use case, this action scans _key comments_ and will set up extra Git and SSH configuration to make things work.
 
-1. When creating the deploy key for a repository like `git@github.com:owner/repo.git` or `https://github.com/owner/repo`, put that URL into the key comment.
+1. When creating the deploy key for a repository like `git@github.com:owner/repo.git` or `https://github.com/owner/repo`, put that URL into the key comment. You can do so using this command: `ssh-keygen -b 4096 -t rsa -N "" -f key -C "git@github.com:my-org/my-swift-package-repo.git"`
```
Comment (mpdude) [inline README.md:71]: It's good to have this hint here :+1:.

At https://github.com/webfactory/ssh-agent#creating-ssh-keys, we show several variants of creating the key. So, maybe we should just have something like 

> You can do so by adding something like `ssh-keygen ... -C "git@github.com:my-org/my-repo.git"` to your `ssh-keygen` command.

Note this section here should have nothing to do with Swift.

File: README.md [hunk 2]
```diff
@@ -68,11 +68,61 @@ When using **Github deploy keys**, GitHub servers will accept the _first_ known
...
 2. After keys have been added to the agent, this action will scan the key comments. 
 3. For key comments containing such URLs, a Git config setting is written that uses [`url.<base>.insteadof`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf). It will redirect `git` requests to URLs starting with either `https://github.com/owner/repo` or `git@github.com:owner/repo` to a fake hostname/URL like `git@...some.hash...:owner/repo`.
 4. An SSH configuration section is generated that applies to the fake hostname. It will map the SSH connection back to `github.com`, while at the same time pointing SSH to a file containing the appropriate key's public part. That will make SSH use the right key when connecting to GitHub.com.
 
+### Support for Swift Packages (using Deploy Key + Secret)
```
Comment (mpdude) [inline README.md:76]: Could you move this to the https://github.com/webfactory/ssh-agent#additional-information-for-particular-tools-or-platforms section?

File: README.md [hunk 3]
```diff
@@ -68,11 +68,61 @@ When using **Github deploy keys**, GitHub servers will accept the _first_ known
...
+  run: |
+	ssh-keyscan github.com >> ~/.ssh/known_hosts
+
+- name: Build Xcode App
+  run: |
+	xcodebuild build -workspace MyCoolApp.xcworkspace -scheme MyCoolAppExample -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 12 Pro,OS=14.4' -scmProvider system
+```
+
+* The important part to note is the use of `-scmProvider system` as it specifies SPM to use the current system's git configuration and **NOT** Xcode's.
+
```
Comment (mpdude) [inline README.md:125]: Puh, it's really hard for me (as I have never used Swift) to judge this section.

My expectation was that this last line (add `-scmProvider system` to `xcodebuild`?) is all you need to know to make this action work.

I mean, isn't the rest about how to set up a private repo in the first place, something people already have when they use this action?

Also, this is only necessary when dealing with multiple deploy keys, right?

**Review by mpdude — CHANGES_REQUESTED** [review]

**Comment by mpdude** [comment]
@cerupcat I know you're using Swift with private dependenies as well. Could you help us to review this additional information for Swift users?

File: README.md (hunk 3 above)
Comment (rr-james-hickman) [reply README.md]: I have moved and consolidated a bit. However, I personally feel like its helpful to have all the steps listed in one spot. It is not always clear what steps are all required, which is why it took 3 various articles (this documentation + and example in an issue + a StackOverflow post) before I was able to correctly get this working. By listing here in one spot, future devs can hopefully save time on getting up and running.

Please feel free to update and commit directly to this branch though.

**Comment by cerupcat** [comment]
I plan to go through the Readme within the next couple of days and try the steps myself to confirm the same thing works for me.

**Comment by rr-james-hickman** [comment]
@cerupcat Any chance you were able to look at this yet?

**Comment by rr-james-hickman** [comment]
@mpdude @cerupcat Bump. Would love to get this closed out and off my PR list if you guys have a few minutes to verify 🙏🏼 
Thanks!

**Comment by rr-james-hickman** [comment]
@mpdude @cerupcat Bumping again 🙏🏼

**Comment by mpdude** [comment]
@rr-james-hickman Thank you James for being so persistent on this one!

As I've mentioned above, I felt like the details you gave in the README for Swift were more than necessary. I was afraid of making the README hard to digest, and also afraid of having too much information in it that might get out of date and was not exactly on scope for this action.

I did, however, pick two small additions and just committed them on `master`, also crediting you.

What do you think – does that do it for you?

Thank you again for your energy put into this!

**Comment by rr-james-hickman** [comment]
@mpdude No worries, glad to be of service. Hopefully future devs can avoid the headaches I had when trying to get everything to work properly in this type of environment setup.

**Pull request closed without merging**